llvmlite/ir/instructions.py

```diff
@@ -95,18 +98,40 @@ def called_function(self):
         """Alias for llvmpy"""
         return self.callee
 
-    def descr(self, buf):
+    def _descr(self, buf, add_metadata):
         args = ', '.join('{0} {1}'.format(a.type, a.get_reference())
                          for a in self.args)
         fnty = self.callee.type
         callee_ref = "{0} {1}".format(fnty, self.callee.get_reference())
         if self.cconv:
             callee_ref = "{0} {1}".format(self.cconv, callee_ref)
-        print("{tail}call {callee}({args}){metadata}".format(
+        print("{tail}{opname} {callee}({args}){attributes}{metadata}".format(
             tail='tail ' if self.tail else '',
+            opname=self.opname,
             callee=callee_ref,
             args=args,
-            metadata=self._stringify_metatdata(),
+            attributes=''.join([" " + attr for attr in self.attributes]),
+            metadata=self._stringify_metatdata() if add_metadata else "",
+            ), file=buf)
+
+    def descr(self, buf):
+        self._descr(buf, add_metadata=True)
+
+
+class InvokeInstr(CallInstr):
+    def __init__(self, parent, func, args, normal_to, unwind_to, name='', cconv=None):
+        assert isinstance(normal_to, Block)
+        assert isinstance(unwind_to, Block)
+        super(InvokeInstr, self).__init__(parent, func, args, name, cconv)
+        self.opname = "invoke"
+        self.normal_to = normal_to
+        self.unwind_to = unwind_to
+
+    def descr(self, buf):
+        super(InvokeInstr, self)._descr(buf, add_metadata=False)
```

Comment:
This simply ignores any metadata that's attached to the instruction? Why not let the metadata through, if any? If "invoke" doesn't accept any metadata, then attaching metadata to it would be a programming error.